genapi/genapi.go

```diff
@@ -692,11 +702,36 @@ func (g *GenAPI) doLever() {
 	g.Lever.Parse()
 }
 
+func (g *GenAPI) srvClientPreprocess(m *dns.Msg) {
+	dc := g.getDCHash()
+	if dc == "" {
+		return
+	}
+	for i := range m.Answer {
+		if ansSRV, ok := m.Answer[i].(*dns.SRV); ok {
+			tar := ansSRV.Target
+			if dc == tar[:len(dc)] {
+				ansSRV.Priority = uint16(4)
```

Comment:
I know I told you to do this but its kinda iffy now that I think about it... I'm tempted to just say subtract one (or even two) instead. We're just randomly adding a `4` here... but like technically this could be getting SRV records from anywhere.
We could also just filter out results from other DCs instead and if there are none left then add the original ones back?